starlite/testing.py

```diff
@@ -280,6 +281,188 @@ def test_my_handler() -> None:
     )
 
 
+class RequestFactory:
+    def __init__(
+        self,
+        app: Starlite = Starlite(route_handlers=[]),
+        server: str = "test.org",
+        port: int = 3000,
+        root_path: str = "",
+    ):
+        self.app = app
+        self.server = server
+        self.port = port
+        self.root_path = root_path
+
+    def _create_scope(
+        self,
+        path: str,
+        http_method: HttpMethod,
+        user: Any = None,
+        auth: Any = None,
+    ) -> Dict[str, Any]:
+        return dict(
+            type="http",
+            method=http_method,
+            server=(self.server, self.port),
+            root_path=self.root_path.rstrip("/"),
+            path=path,
+            headers=[],
+            app=self.app,
+            user=user,
+            auth=auth,
+        )
+
+    @classmethod
+    def _create_cookie_header(
+        cls, headers: Dict[str, str], cookies: Optional[Union[List["Cookie"], str]] = None
+    ) -> None:
+        if not cookies:
+            return None
+
+        if isinstance(cookies, list):
+            cookie_header = "; ".join(cookie.to_header(header="") for cookie in cookies)
+            headers[ParamType.COOKIE] = cookie_header
+        elif isinstance(cookies, str):
+            headers[ParamType.COOKIE] = cookies
+
+    def _build_headers(
+        self,
+        headers: Optional[Dict[str, str]] = None,
+        cookies: Optional[Union[List["Cookie"], str]] = None,
+    ) -> List[Tuple[bytes, bytes]]:
+        headers = headers or {}
+        self._create_cookie_header(headers, cookies)
+        return [
+            ((key.lower()).encode("latin-1", errors="ignore"), value.encode("latin-1", errors="ignore"))
+            for key, value in headers.items()
+        ]
+
+    def _create_request_with_data(
+        self,
+        http_method: HttpMethod,
+        path: str,
+        headers: Optional[Dict[str, str]] = None,
+        cookies: Optional[Union[List["Cookie"], str]] = None,
```

Comment:
This is where we can create a dictionary from cookies and pass it to `session`. See the comment above.